README.adoc

```diff
@@ -1,5 +1,256 @@
+this issue the `@ContextAccessor` annotation and a `bindContext` method.
+
+Example:
+[source,java]
+----
+@TemplateHelper
+@ContextAccessor
+public class StoredVariableHelper extends StaticMethodValueResolver {
+
+    public static void bindContext(Map<String, ?> context) {
+        ThreadLocalContextHolder.bindContext(context);
+    }
+
+    public static synchronized String getApiPrefixLocal(Object object) {
+        return (String) ThreadLocalContextHolder.getVariable("apiPrefix");
+    }
+
+    public static synchronized Boolean isGenerateOptionalTypes(Object object) {
+        return Boolean.parseBoolean((String) ThreadLocalContextHolder.getVariable("generateOptionalTypes"));
+    }
+
+    public static synchronized Boolean isGeneratePayloadValidator(Object object) {
+        return Boolean.parseBoolean((String) ThreadLocalContextHolder.getVariable("generatePayloadValidator"));
+    }
+}
+
+----
+
+This only need if we used the templateParameters in a java Helper.
+
+=== Checksum generation
+A checksum is a value calculated from data to detect errors during transmission or storage. It ensures data integrity and any mismatches indicate errors in the data.
+
+* The checksum generation is a goal of the generator-maven-plugin.
+
+Example:
+
+[source,xml]
+----
+                    <execution>
+                        <id>execute-ui-checksum-from-file</id>
+                        <phase>generate-test-resources</phase>
+                        <goals>
+                            <goal>checksum</goal>
+                        </goals>
+                        <configuration>
+                            <ui>${basedir}/src/test/resources/MultiPrincipalTest-ui.model</ui>
+                            <destination>${basedir}/target/test-classes/ui/file</destination>
+                        </configuration>
+                    </execution>
+----
+* The [modelName]ProjectGenerateMojo is responsible for generating the files, while the [modelName]ProjectCalculateChecksumMojo is responsible for generating checksums for the files.
+
+* The .generated-files-[actor]-[actor] contains the actorTypeBased checksums. The .generated-files contains the not actorTypeBased checksums
+
+** A line contains a file name and a md5sum value.
+
+=== Generator ignore file
+
+IMPORTANT: Sometimes a developer needs to replace generated file with custom developed file. On that case the generator has to ignore the given file to keep the edited version. To achieve this `.generator-ignore` file can be used. In this case, every unexpected behavior is the responsibility of the developer.
+
+We use GLOB format, so the usage is same as .gitignore.
+
+Here is some example :
+
+* Ignore files with a specific extension: *.log, *.txt, *.docx
+
+* Ignore a specific file: config.ini, secrets.txt
+
+* Ignore a specific directory: node_modules/
```

Comment:
We should replace the `node_modules` example, because it's not generated, but present in `.gitignore`

Reply:
[b8f2188](https://github.com/BlackBeltTechnology/judo-generator-commons/pull/29/commits/b8f218814caf482868a6d80c23e83c6ca810567d)